From d57b8833c43d297004d8393d8e54a2ab2655a899 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Fri, 3 May 2013 17:16:05 +0100
Subject: [PATCH] usrmount: Fix miss-detection of btrfs subvolumes.

This causes the root FS options to be incorrectly applied to to /usr
In some cases this can cause boot failure e.g. due to and XFS /usr
not supporting the 'acl' option from the ext4 root FS.

https://bugs.mageia.org/show_bug.cgi?id=9884
---
 modules.d/98usrmount/mount-usr.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/98usrmount/mount-usr.sh b/modules.d/98usrmount/mount-usr.sh
index a4a66ae..de7dcc2 100755
--- a/modules.d/98usrmount/mount-usr.sh
+++ b/modules.d/98usrmount/mount-usr.sh
@@ -61,14 +61,14 @@ mount_usr()
                 LABEL=*)
                     _dev="$(echo $_dev | sed 's,/,\\x2f,g')"
                     _dev="/dev/disk/by-label/${_dev#LABEL=}"
-		    ;;
+                    ;;
                 UUID=*)
                     _dev="${_dev#block:}"
                     _dev="/dev/disk/by-uuid/${_dev#UUID=}"
                     ;;
             esac
             if strstr "$_opts" "subvol=" && \
-                [ "${root#block:}" -ef $_dev ]
+                [ "${root#block:}" -ef $_dev ] && \
                 [ -n "$rflags" ]; then
                 # for btrfs subvolumes we have to mount /usr with the same rflags
                 rflags=$(filtersubvol "$rflags")
